web/src/hooks/arrows.test.ts

```diff
@@ -0,0 +1,85 @@
+import { describe, expect, it } from 'vitest';
+
+import { filterExchanges } from './arrows';
+
+const mockExchangesToExcludeZoneView = ['exchange1', 'exchange2', 'exchange4'];
+
+const mockExchangesToExcludeCountryView = ['exchange3', 'exchange5'];
+
+const mockExchangesResponses = {
+  exchange1: {
+    '2024-01-31T12:00:00Z': { co2intensity: 137.68, netFlow: -44 },
+    '2024-01-31T13:00:00Z': { co2intensity: 133.44, netFlow: -57 },
+  },
+  exchange2: {
+    '2024-01-31T14:00:00Z': { co2intensity: 131.44, netFlow: -56 },
+    '2024-01-31T15:00:00Z': { co2intensity: 135.82, netFlow: -55 },
+  },
+  exchange3: {
+    '2024-01-31T16:00:00Z': { co2intensity: 168.29, netFlow: -55 },
+    '2024-01-31T17:00:00Z': { co2intensity: 199.8, netFlow: -13 },
+  },
+  exchange4: {
+    '2024-01-31T18:00:00Z': { co2intensity: 199.73, netFlow: -18 },
+    '2024-01-31T19:00:00Z': { co2intensity: 195.2, netFlow: -18 },
+  },
+  exchange5: {
+    '2024-01-31T20:00:00Z': { co2intensity: 188.22, netFlow: -11 },
+    '2024-01-31T21:00:00Z': { co2intensity: 178.86, netFlow: -50 },
+  },
+};
```

Comment:
FYI I used GitHub Copilot to generate this test data after giving it a real snippet. Really handy.